planner/cascades/transformation_rules.go

```diff
@@ -953,10 +925,68 @@ func (r *PushSelDownJoin) OnTransform(old *memo.ExprIter) (newExprs []*memo.Grou
 	}
 	leftCond = expression.RemoveDupExprs(sctx, leftCond)
 	rightCond = expression.RemoveDupExprs(sctx, rightCond)
+	return
+}
+
+// PushSelDownJoin pushes Selection through Join.
+type PushSelDownJoin struct {
+	baseRule
+	pushDownJoin
+}
+
+// NewRulePushSelDownJoin creates a new Transformation PushSelDownJoin.
+// The pattern of this rule is `Selection -> Join`.
+func NewRulePushSelDownJoin() Transformation {
+	rule := &PushSelDownJoin{}
+	rule.pattern = memo.BuildPattern(
+		memo.OperandSelection,
+		memo.EngineTiDBOnly,
+		memo.NewPattern(memo.OperandJoin, memo.EngineTiDBOnly),
+	)
+	return rule
+}
+
+// Match implements Transformation interface.
+func (r *PushSelDownJoin) Match(expr *memo.ExprIter) bool {
+	return !expr.GetExpr().HasAppliedRule(r)
+}
+
+// buildChildSelectionGroup builds a new childGroup if the pushed down condition is not empty.
+func buildChildSelectionGroup(
+	sctx sessionctx.Context,
+	blockOffset int,
+	conditions []expression.Expression,
+	childGroup *memo.Group) *memo.Group {
+	if len(conditions) == 0 {
+		return childGroup
+	}
+	newSel := plannercore.LogicalSelection{Conditions: conditions}.Init(sctx, blockOffset)
+	groupExpr := memo.NewGroupExpr(newSel)
+	groupExpr.SetChildren(childGroup)
+	newChild := memo.NewGroupWithSchema(groupExpr, childGroup.Prop.Schema)
+	return newChild
+}
+
+// OnTransform implements Transformation interface.
+// This rule tries to pushes the Selection through Join. Besides, this rule fulfills the `XXXConditions` field of Join.
+func (r *PushSelDownJoin) OnTransform(old *memo.ExprIter) (newExprs []*memo.GroupExpr, eraseOld bool, eraseAll bool, err error) {
+	sel := old.GetExpr().ExprNode.(*plannercore.LogicalSelection)
+	joinExpr := old.Children[0].GetExpr()
+	join := joinExpr.ExprNode.(*plannercore.LogicalJoin)
+	newJoin := join.Shallow()
+	sctx := sel.SCtx()
+	leftGroup := old.Children[0].GetExpr().Children[0]
+	rightGroup := old.Children[0].GetExpr().Children[1]
+
+	leftCond, rightCond, remainCond, dual := r.predicatePushDownJoin(sctx, sel.Conditions, newJoin, leftGroup.Prop.Schema, rightGroup.Prop.Schema)
+	if dual != nil {
+		return []*memo.GroupExpr{memo.NewGroupExpr(dual)}, true, true, nil
+	}
+
 	// TODO: Update EqualConditions like what we have done in the method join.updateEQCond() before.
-	leftGroup = buildChildSelectionGroup(sel, leftCond, leftGroup)
-	rightGroup = buildChildSelectionGroup(sel, rightCond, rightGroup)
-	newJoinExpr := memo.NewGroupExpr(join)
+	leftGroup = buildChildSelectionGroup(sel.SCtx(), sel.SelectBlockOffset(), leftCond, leftGroup)
```

Comment:
We need to consider the join type here. For example:
`select * from t1 left join t2 on t1.a > 10`, the condition `t1.a > 10` cannot be pushed down.

Reply:
if is right join, `len(leftCond) == 0`?.I'm not sure.I will add a test for it.
Ok, I will wait for  #15426.May be We should `TransformApplyCondToSel` too.